core/templates/dev/head/pages/exploration_editor/exploration_metadata_modal_directive.html

```diff
@@ -6,11 +6,10 @@
   <p>
     Just a few more things before you publish:
   </p>
-  <p ng-if="requireTitleToBeSpecified">
-    Add a title: what should your exploration be called?
-    <input type="text" class="form-control" ng-model="explorationTitleService.displayed"
-           placeholder="Music in the 19th Century" maxlength="40">
-  </p>
+  <title-field
```

Comment:
I see what you're saying. I think more fields imposes more cognitive load, but it is unclear how significant the effect is.

That said, if we want to change the actual design, let's do that separately from this PR, especially since it's controversial -- the current PR should maintain the existing behaviour, since it's supposed to be a refactor with no user-visible effects. I recommend consulting Jared if you'd like to show all the fields, and if it's a good idea then we can file a separate issue for it.